src/Utility/HDF5Reader.C

```diff
@@ -120,6 +124,43 @@ bool HDF5Reader::readString (const std::string& name, std::string& out)
 }
 
 
+bool HDF5Reader::read3DArray (const std::string& name, Matrix3D& out)
+{
+#ifdef HAS_HDF5
+  if (!this->openFile(H5F_ACC_RDONLY))
+    return false;
+
+  hid_t set = H5Dopen2(m_file,name.c_str(),H5P_DEFAULT);
+  hid_t space = H5Dget_space(set);
+  hsize_t dims = H5Sget_simple_extent_ndims(space);
+
+  if (dims != 3)
+    return false;
+
+  std::array<hsize_t, 3> ndim;
```

Comment:
i see neither a problem with, nor any reason to ,change to old style array C array.

Reply:
ok, keep the array. My only thought was that since `hsize_t ndim[3]` is so much shorter (and don't require the extra include), there was a particular reason for using std::array here instead.